File: Java/Math/Number of divisors.java
//https://www.geeksforgeeks.org/problems/number-of-divisors1631/1?
class Solution {
    static int countDivisors(int n) {
        // code here
        int count = 0;
        
        for (int i = 1; i <= Math.sqrt(n); i++) {
            int k = n / i;
            
            if (n % i == 0) {
                if (i % 3 == 0) {
                    count++;
                    // System.out.println(i);
                }
                if (k % 3 == 0 && k != i) {
                    // System.out.println(k);
                    count++;
                }
            }
        }
        return count;
    }
}

File: Python/Math/Number of divisors.py
//https://www.geeksforgeeks.org/problems/number-of-divisors1631/1?
class Solution:
    def countDivisors(self, n):
        # code here
        return sum(n%i == 0 and i%3 == 0 for i in range(1, N + 1))
